### PR TITLE
Feature Branch - Implementation more menu for Dashboard Cards

### PR DESCRIPTION
**Pull request by AjeshRPai**
## Description
This PR  
1. Adds the more menu to all the dashboard cards 

| Card  | More menu |
|---|---|
| Activity Log Card | |
| Pages Card| |
| Today Stats Card | |   
| Blaze Card(Promote) | |
| Blaze Card(Campaign) | | 
| Draft Posts Card | |
| Scheduled Posts Card |  |
| Quick Start Card|   |

3. Adds **hide this** option to all the dashboard cards 
4. Adds more menu options to all the dashboard cards 
5. Removes footer from Today's stats, Activity log, Draft Post Card and Scheduled Post card

## Not included in the PR 
The feature to unhide the cards once they are hidden is in progress. 

## To test:
1. Login to the Jetpack app 
3. Go to MySiteDashboard 
4. Verify the functionality of each card 
5. Click on the more menu of each card and verify that it is working as expected
7. Click on "hide this" for each card and verify that the card is hidden for the site 

## This PR includes the following PR's 
- https://github.com/wordpress-mobile/WordPress-Android/pull/18968
- https://github.com/wordpress-mobile/WordPress-Android/pull/19005
- https://github.com/wordpress-mobile/WordPress-Android/pull/19008
- https://github.com/wordpress-mobile/WordPress-Android/pull/19112
- https://github.com/wordpress-mobile/WordPress-Android/pull/19011
- https://github.com/wordpress-mobile/WordPress-Android/pull/18994
- https://github.com/wordpress-mobile/WordPress-Android/pull/19020
- https://github.com/wordpress-mobile/WordPress-Android/pull/19055
- https://github.com/wordpress-mobile/WordPress-Android/pull/19042
- https://github.com/wordpress-mobile/WordPress-Android/pull/19043
- https://github.com/wordpress-mobile/WordPress-Android/pull/19047
- https://github.com/wordpress-mobile/WordPress-Android/pull/19053
- https://github.com/wordpress-mobile/WordPress-Android/pull/19058
- https://github.com/wordpress-mobile/WordPress-Android/pull/19059
- https://github.com/wordpress-mobile/WordPress-Android/pull/19056
- https://github.com/wordpress-mobile/WordPress-Android/pull/19063
- https://github.com/wordpress-mobile/WordPress-Android/pull/19099
- https://github.com/wordpress-mobile/WordPress-Android/pull/19075
- https://github.com/wordpress-mobile/WordPress-Android/pull/19089
- https://github.com/wordpress-mobile/WordPress-Android/pull/19090
- https://github.com/wordpress-mobile/WordPress-Android/pull/19091
- https://github.com/wordpress-mobile/WordPress-Android/pull/19104
- https://github.com/wordpress-mobile/WordPress-Android/pull/19107

## ⚠️  Merge Instructions 
- Merge FLUXC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2814
- Update the FluxC Hash version 
- Remove Donot Merge label
- Wait for CI to Complete
- Merge as normal

Note: Since this PR implements the Hide functionality only, and Unhide functionality is in Progress. Please wait untill the release branch is cut for 23.2 to merge this PR

## Regression Notes
1. Potential unintended areas of impact
- Dashboard cards are not working as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and Manual tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by ravishanker — APPROVED** [review]
Great work 👍 

Thanks for taking on a mammoth task of decomposing MySite into manageable chunks, and help avoid bugs. 👏

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/activity/ActivityLogCardViewModelSlice.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.wordpress.android.ui.mysite.SelectedSiteRepository
+import org.wordpress.android.ui.mysite.SiteNavigationAction
+import org.wordpress.android.ui.mysite.cards.dashboard.CardsTracker
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ActivityLogCardViewModelSlice @Inject constructor(
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/activity/ActivityLogCardViewModelSlice.kt:16]: Naming could just use ActivityLogCardViewModel without the suffix **Slice** perhaps!  Everywhere.

Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/activity/ActivityLogCardViewModelSlice.kt:16]: Hey @ravishanker 
Thanks for that suggestion. When we took on the task I wasn't sure what to name it to be honest. We will re-visit the naming again and change it in a future PR 👍🏼

**Comment by AjeshRPai** [comment]
Thanks for the review @ravishanker 🙌🏼

**Pull request merged**